ext/Kbuild

```diff
@@ -1,3 +1,5 @@
+subdir-ccflags-y += -D__ZEPHYR_KERNEL__
```

Comment:
we can't make this assumption, some code in ext/ might not be possible to host under lib/ and has to be under ext/ and be userspace. I suggest to do that on a per component basis (not per file basis)